ui/components/src/model/units/decimal.ts

```diff
@@ -33,50 +32,24 @@ export const DECIMAL_UNIT_CONFIG: Readonly<Record<DecimalUnitKind, UnitConfig>>
   },
 };
 
-export function formatDecimal(value: number, unitOptions: DecimalUnitOptions): string {
-  const decimalPlaces = unitOptions.decimal_places ?? DEFAULT_DECIMAL_PLACES;
+export function formatDecimal(value: number, { abbreviate, decimal_places }: DecimalUnitOptions): string {
+  decimal_places = decimal_places ?? DEFAULT_DECIMAL_PLACES;
 
-  if (value === 0) {
-    return value.toString();
+  // Avoids maximumFractionDigits value is out of range error. Possible values are 0 to 20.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#minimumfractiondigits
+  if (decimal_places < 0) {
+    decimal_places = 0;
+  } else if (decimal_places > 20) {
+    decimal_places = 20;
   }
 
-  if (unitOptions.abbreviate && value >= 1000) {
-    return abbreviateLargeNumber(value, decimalPlaces);
-  }
+  const showFullNumber = abbreviate == false || value < 1000;
 
-  const formatParams: Intl.NumberFormatOptions = {
+  const formatter = new Intl.NumberFormat('en-US', {
     style: 'decimal',
-    maximumFractionDigits: decimalPlaces,
+    notation: showFullNumber ? 'standard' : 'compact',
+    maximumFractionDigits: decimal_places,
     useGrouping: true,
-  };
-  const decimalFormatter = new Intl.NumberFormat('en-US', formatParams);
-  return decimalFormatter.format(value);
-}
-
-/**
- * Takes large numbers and abbreviates them with the appropriate suffix
- * 10123 -> 10.123k
- * 1000000 -> 1M
- */
-export function abbreviateLargeNumber(num: number, decimalPlaces = 2) {
```

Comment:
Trying to remember, but we might be using this function somewhere internally btw

Reply:
Nvm just checked, we aren't importing this one anywhere! There is a similar `abbreviateLargeNumber` function in that repo that I was thinking of

Reply:
I just deleted them without thinking if they were used outside of this file :)